README for run operator (#14)
wherobots/airflow-providers-wherobots: README.md

$ pip install airflow-providers-wherobots
```

### Create an http connection

> You first need to create a Connection in Airflow. This can be done from

Reviewer: ```suggestion
Create a Connection in Airflow. This can be done from
```

> the UI, or from the command-line. The default Wherobots connection name

Reviewer: Can it be created from the Wherobots UI or the Airflow UI?

Author: Good point. I have updated it, added a link to the Airlfow document

    --conn-password "$(< api.key)"
```

## Usage

### Execute a `Run` on Wherobots Cloud

> Wherobots allows users to upload their code (.py, .jar),

Reviewer: Is there a limit on or list of the type of file extensions that Wherobots Cloud supports? If so, we should include specifically which file formats are supported.

Author: We don't have such official document yet

run it on the cloud, and monitor the status of the run.

> The `WherobotsRunOperator` allows you to execute a `Run` on Wherobots Cloud.

Reviewer: We should define what a `Run` is for new users.

> It will trigger the run according to the parameters you provide,

Reviewer: ```suggestion
`WherobotsRunOperator` triggers the run according to the parameters you provide,
```

> and will wait for the run to finish before completing the task.

Reviewer: ```suggestion
and waits for the run to finish before completing the task.
```

Refer to the [Wherobots Managed Storage Documentation](https://docs.wherobots.com/latest/develop/storage-management/storage/)
to learn more about how to upload and manage your code on Wherobots Cloud.

> Below is an example `WherobotsRunOperator`

Reviewer: ```suggestion
Below is an example of `WherobotsRunOperator`
```

```python
operator = WherobotsRunOperator(
        task_id="your_task_id",
        name="airflow_operator_test_run_{{ ts_nodash }}",
        run_python={
            "uri": "s3://wbts-wbc-m97rcg45xi/42ly7mi0p1/data/shared/classification.py"
        },
        runtime="tiny-a10-gpu",
        dag=dag,
        poll_logs=True,
    )
```

#### Arguments

The arguments for the `WherobotsRunOperator` constructor:

The `run_*` arguments are mutually exclusive, you can only specify one of them.

* `name: str`: The name of the run.
  If not specified, a default name will be
  generated.

> * `runtime: str`: The runtime decides the size of the workloads that execute the run.

Reviewer 2: Note: I just realized this is for the readme, not the documentation. :grimacing:

> * `runtime: str`: The runtime decides the size of the workloads that execute the run.

Author: This is a good point. I'm working on optimizing the runtime value options and add an official doc for that

  The default value is `30`.
* `run_python: dict`: A dictionary with the following keys:
  * `uri: str`: The URI of the Python file to run.
  * `args: list[str]`: A list of arguments to pass to the Python file.
* `run_jar: dict`: A dictionary with the following keys:
  * `uri: str`: The URI of the JAR file to run.
  * `args: list[str]`: A list of arguments to pass to the JAR file.
  * `mainClass: str`: The main class to run in the JAR file.
* `environment: dict`: A dictionary with the following keys:
  * `sparkDriverDiskGB: int`: The disk size for the Spark driver.
  * `sparkExecutorDiskGB: int`: The disk size for the Spark executor.
  * `sparkConfigs: dict`: A dictionary of Spark configurations.
  * `dependencies: list[dict]`: A list of dependant libraries to install.

The `dependencies` argument is a list of dictionaries.
There are two types of dependencies supported.

1. `PYPI` dependencies:
```json
{
    "sourceType": "PYPI",
    "libraryName": "package_name",
    "libraryVersion": "package_version"
}
```

2. `FILE` dependencies:
```json
{
    "sourceType": "FILE",
    "filePath": "s3://bucket/path/to/dependency.whl"
}
```
File types supported: `.whl`, `.zip`, `.jar`

> ### Execute a SQL query

Reviewer 2: Would it be possible to have the same list of arguments for `WherobotsSqlOperator`?

Author: Sure I'll add separately
